src/libstd/fs/mod.rs

```diff
@@ -73,6 +73,11 @@ pub struct Metadata(fs_imp::FileAttr);
 /// will yield instances of `io::Result<DirEntry>`. Through a `DirEntry`
 /// information like the entry's path and possibly other metadata can be
 /// learned.
+///
+/// # Failure
+///
+/// This `io::Result` will be an `Err` if there's some problem reading the
+/// directory, for example, if it does not exist.
```

Comment:
I think this may not be quite the error that we're looking for because a nonexistent directory should return an error from the initial `read_dir` function call, not halfway through iteration. I haven't been able to find some great documentation on when this can fail or not, but I think it's fine to just say something about intermittent I/O errors perhaps.